File: app/src/main/java/com/iug/coursehub/feature/user/home_screen/course_categories_screen/CourseCategoriesViewModel.java
package com.iug.coursehub.feature.user.home_screen.course_categories_screen;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class CourseCategoriesViewModel extends ViewModel {

    private final MutableLiveData<UiAction> _uiAction = new MutableLiveData<>();
    public final LiveData<UiAction> uiAction = _uiAction;

    public void navigateToCourseDetails(int courseId) {
        _uiAction.postValue(new UiAction.NavigateToCourseDetails(courseId));
    }

    public interface UiAction {
        class NavigateToCourseDetails implements UiAction {
            private final int courseId;

            public NavigateToCourseDetails(int courseId) {
                this.courseId = courseId;
            }

            public int getCourseId() {
                return courseId;
            }
        }
    }
}

File: app/src/main/java/com/iug/coursehub/feature/user/home_screen/cuorse_categories_screen/CourseCategoriesViewModel.kt
package com.iug.coursehub.feature.user.home_screen.cuorse_categories_screen

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class CourseCategoriesViewModel : ViewModel() {

    private val _uiAction = MutableLiveData<UiAction>()
    val uiAction: LiveData<UiAction> = _uiAction

    fun navigateToCourseDetails(courseId: Int) {
        _uiAction.postValue(UiAction.NavigateToCourseDetails(courseId))
    }

    sealed interface UiAction {
        data class NavigateToCourseDetails(val courseId: Int) : UiAction
    }
}